### DISCUSSION TITLE
Big Parsers?

### DISCUSSION THREAD

**Discussion by cjhowe-us** [peter-winter/ctpg#68] [Q&A]
Hey, thanks for writing this library. Is ctpg capable of parsing complex languages like C/C++? Have there been any open source big parsers with many rules and terms built on top of ctpg that I could refer to?

**Comment by peter-winter** [comment]
This is not really a question to me. Noone has ever told me where they are using ctpg.
I suspect really big parsers are a problematic beacues of compilation time / ram usage by compiler.
I'm trying to optimize the library as I can, although lack of time is a big problem lately.

The branch I'm working on (term-groups) is a big step forward, it basically allows you to group similar operators, like the 11 assignment operators in C. From parser point of view they are the same. This greatly reduces the resources compiler uses to compile ctpg based parsers.

**Comment by haasdo95** [comment]
I am currently working on a compile-time parser & plan generator of SQL. 
Indeed, noticing very long compile time...
Any tips on reducing the compile time at the moment?...
Anyway, looking forward to the new release!
All the best.

**Reply by peter-winter** [reply]
Without seeing your code it is hard for me to give you any tips. 
I'm really struggling with time to make the new realease possible.

**Reply by haasdo95** [reply]
Thanks for your reply!
Here's my parser for (a small subset of) SQL. https://github.com/haasdo95/ctsql/tree/master/include/parser
There's no real hurry so take ur time!...
